src/main/java/net/fabricmc/loom/task/RemapJarTask.java

```diff
@@ -176,8 +197,20 @@ public void scheduleRemap() throws Throwable {
 
 		MappingsProvider mappingsProvider = extension.getMappingsProvider();
 
-		String fromM = "named";
-		String toM = "intermediary";
+		String fromM = this.fromM.get();
+		String toM = this.toM.get();
+
+		TinyTree mappings = mappingsProvider.getMappings();
+		List<String> namespaces = mappings.getMetadata().getNamespaces();
+
+		// Validate the namespaces are supported in the tiny tree
+		if (!namespaces.contains(fromM)) {
+			RemapJarTask.throwMissingNamespaceException(project, fromM, namespaces);
+		}
+
+		if (!namespaces.contains(toM)) {
+			RemapJarTask.throwMissingNamespaceException(project, toM, namespaces);
+		}
```

Comment:
```suggestion
		if (!namespaces.contains(fromM)) {
			throwMissingNamespaceException(project, fromM, namespaces);
		}

		if (!namespaces.contains(toM)) {
			throwMissingNamespaceException(project, toM, namespaces);
		}
```
There's no need to specify the class name here.

Reply:
I qualify these things so I know whether a method is static or belongs to a class file (usually via `this`)

Reply:
Of course no conventions exist on whether member references should be qualified or not